File: javascript/grade-school/grade_school.js
var roster;

function School() {
  roster = {};
}

School.prototype.roster = function() {
  return JSON.parse(JSON.stringify(roster));
}

School.prototype.add = function (name, grade) {
  roster[grade] = this.grade(grade).concat(name).sort();
}

School.prototype.grade = function (grade) {
  return JSON.parse(JSON.stringify(roster[grade] || []));
}

module.exports = School;

File: elixir/grade-school/grade_school.exs
defmodule School do

  def add(db, name, grade) do
    Map.update(db, grade, [ name ], &( [ name | &1 ] ))
  end

  def grade(db, grade) do
    db |> Map.get(grade, [])
       |> Enum.sort
  end

  def sort(db) do
    db |> Enum.map(fn { key, _ } -> { key, grade(db, key) } end)
       |> Enum.sort
  end

end
